pkg/plugin/plugin_test.go

```diff
@@ -317,3 +340,83 @@ func TestCreateOrUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestNewPlugin(t *testing.T) {
+	log := logrus.NewEntry(logrus.New())
+	tests := map[string]struct {
+		f            func(*api.PluginConfig)
+		expectedErrs []error
+	}{
+		"empty syncImage": {
+			f: func(p *api.PluginConfig) {
+				p.SyncImage = ""
+			},
+			expectedErrs: []error{
+				errors.New(`syncImage cannot be empty`),
+			},
+		},
+		"empty imagePullSecret": {
+			f: func(p *api.PluginConfig) {
+				p.AzureConfig.ImagePullSecret = ""
+			},
+			expectedErrs: []error{
+				errors.New(`imagePullSecret cannot be empty`),
+			},
+		},
+		"nil loggingCert.Key": {
+			f: func(p *api.PluginConfig) {
+				p.AzureConfig.LoggingCert.Key = nil
+			},
+			expectedErrs: []error{
+				errors.New(`loggingCert key cannot be nil`),
+			},
+		},
+		"nil loggingCert.Cert": {
+			f: func(p *api.PluginConfig) {
+				p.AzureConfig.LoggingCert.Cert = nil
+			},
+			expectedErrs: []error{
+				errors.New(`loggingCert certificate cannot be nil`),
+			},
+		},
+	}
+
+	for name, test := range tests {
+		pCfg, err := getDummyPluginConfig()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		test.f(pCfg)
+		_, err = NewPlugin(log, pCfg)
+
+		if !reflect.DeepEqual(err, kerrors.NewAggregate(test.expectedErrs)) {
+			t.Errorf("%s expected errors:", name)
+			t.Errorf("%s != %s", err, test.expectedErrs)
+		}
+	}
+}
+
+func getDummyPluginConfig() (*api.PluginConfig, error) {
+	// Generate dummy certs
+	// TODO: This is expensive. Replace with dummy certs from tls testdata
```

Comment:
Will do this if we agree this is fine approach to add error to `NewPlugin`

Reply:
grep the codebase for dummyPrivateKey and consolidate please